server/app/dal/Dao.scala

```diff
@@ -62,6 +72,11 @@ class DaoImpl @Inject()(cluster: CassandraCluster)(implicit ec: ExecutionContext
       val limitInt: Integer = limit
       session.executeAsync(getLastTasksStatement.bind(year(yearAgo), `type`.toString, limitInt))
     }.map(allTasks))
+
+  def getTask(year: Long, `type`: TaskType, timeuuid: UUID): Future[Option[Task]] = TryFuture(
+    toFuture {
+      session.executeAsync(getTaskStatement.bind(new Timestamp(year), `type`.toString, timeuuid))
+    }.map(rs => if (rs.iterator().hasNext) Option(toTask(rs.one)) else Option.empty))
```

Comment:
Let's do it more functional way - https://github.com/DmytroOrlov/devgym/commit/867041ea0c89aa481d1181a0d0a365b22bf100f1